summingbird-storm/src/main/scala/com/twitter/summingbird/storm/BaseBolt.scala

```diff
@@ -151,11 +151,13 @@ case class BaseBolt[I, O](jobID: JobId,
   }
 
   private def finish(inputs: Chain[InputState[Tuple]], results: TraversableOnce[O]) {
-    val tuples = inputs.iterator.map(_.state).toList
+    var numTuples = Option.empty[Int]
     var emitCount = 0
     if (hasDependants) {
       if (anchorTuples.anchor) {
         results.foreach { result =>
+          val tuples = inputs.iterator.map(_.state).toList
```

Comment:
don't we want this above the foreach? We don't want to recompute `tuples` for each result. We just want it once and then reuse for all results, no?